### PR TITLE
Refactor L10n for testing

### PR DESCRIPTION
**Pull request by nupplaphil**
- Moving L10n to L10n\L10n
- Adding constructor information
- Adding to factory
- simplify/speedup tests

1. This is one prerequisite for some followup improvements for testing.
2. Another PR will add some unit-tests for L10n as well

- Local node works
- Changing language in profile works
- Working during installation (found bug during test see #7358 )

### CODE REVIEW

**Comment by nupplaphil** [comment]
btw. This PR will decrease the test duration about 2 min per test :)

File: src/Core/Update.php [hunk 1]
```diff
@@ -252,7 +252,7 @@ private static function updateFailed($update_id, $error_message) {
 			$sent[] = $admin['email'];
 
 			$lang = (($admin['language'])?$admin['language']:'en');
-			L10n::pushLang($lang);
+			 pushLang($lang);
```
Comment (MrPetovan) [inline src/Core/Update.php]: Huh?

File: tests/include/ApiTest.php [hunk 2]
```diff
@@ -60,7 +61,11 @@ public function setUp()
 		Factory\ConfigFactory::createPConfig($configCache);
 		$logger = Factory\LoggerFactory::create('test', $database, $config, $profiler);
 		$baseUrl = new BaseURL($config, $_SERVER);
-		$this->app = new App($database, $config, $mode, $router, $baseUrl, $logger, $profiler, false);
+		$l10n = new L10n(L10n::detectLanguage($config->get('system', 'language', 'de')),
```
Comment (MrPetovan) [inline tests/include/ApiTest.php]: `de` by default, really?

File: src/Factory/DependencyFactory.php [hunk 3]
```diff
@@ -36,7 +37,10 @@ public static function setUp($channel, $directory, $isBackend = true)
 		$logger = Factory\LoggerFactory::create($channel, $database, $config, $profiler);
 		Factory\LoggerFactory::createDev($channel, $config, $profiler);
 		$baseURL = new BaseURL($config, $_SERVER);
+		$l10n = new L10n(L10n::detectLanguage($config->get('system', 'language', 'de')),
```
Comment (MrPetovan) [inline src/Factory/DependencyFactory.php]: `de` by default, really?

**Comment by nupplaphil** [comment]
temporary closed until ready :)

**Pull request closed without merging**